Lib/test/_test_multiprocessing.py

```diff
@@ -2712,6 +2712,27 @@ def errback(exc):
         p.close()
         p.join()
 
+    def test_pool_worker_died_without_communicating(self):
+        # Issue22393: test fix of indefinite hang caused by worker processes
+        # exiting abruptly (such as via os._exit()) without communicating
+        # back to the pool at all.
+        prog = (
+            "import os, multiprocessing as mp; "
+            "is_main = (__name__ == '__main__'); "
+            "p = mp.Pool(1) if is_main else print('worker'); "
+            "p.map(os._exit, [1]) if is_main else None; "
+            "(p.close() or p.join()) if is_main else None"
+        )
+        # Only if there is a regression will this ever trigger a
+        # subprocess.TimeoutExpired.
+        completed_process = subprocess.run(
+            [sys.executable, '-E', '-S', '-O', '-c', prog],
```

Comment:
The '-O' flag probably shouldn't be used here, but '-S' and '-E' seem fine.

Reply:
Also, consider calling `test.support.script_utils.interpreter_requires_environment()`, and only use the '-E' flag if that returns `False`, as done by the other Python script running utils in `test.support.script_utils`.

Or just use `test.support.script_utils.run_python_until_end()` instead of `subprocess.run()`.